Resolve remaining Vale warnings

ptgott opened the pull request:
- Add "How it works" headings to 10 how-to guides.
- Remove the architecture-h2 Vale style. We will restore this as a `remark` linter in gravitational/docs-website#271. This linter should have been a `remark` linter to begin with, since otherwise it is not possible to ignore it with MDX comments.
- Disable the upcoming `remark` "How it works" linter for the remaining guides. This lets us decouple the need to add "How it works" sections from the need to get all Vale warnings to pass in order to add an automated Vale workflow that doesn't need to check pull request diffs.
- Disable the former names linter in the Azure AD guide, since we will update this while resolving a GitHub issue.
- Fix various minor warnings.
- Remove the step numbering linter. The plan is to convert this to a remark linter in gravitational/docs-website#271.

ptgott committed:
Resolve remaining Vale warnings

- Add "How it works" headings to 10 how-to guides.
- Remove the architecture-h2 Vale style. We will restore this as a
  `remark` linter in gravitational/docs-website#271. This linter should
  have been a `remark` linter to begin with, since otherwise it is not
  possible to ignore it with MDX comments.
- Disable the upcoming `remark` "How it works" linter for the remaining
  guides. This lets us decouple the need to add "How it works" sections
  from the need to get all Vale warnings to pass in order to add an
  automated Vale workflow that doesn't need to check pull request diffs.
- Disable the former names linter in the Azure AD guide, since we will
  update this while resolving a GitHub issue.
- Fix various minor warnings.
- Remove the step numbering linter. The plan is to convert this to a
  remark linter in gravitational/docs-website#271.

ravicious commented on docs/pages/connect-your-client/model-context-protocol/database-access.mdx:
````diff
@@ -46,7 +46,7 @@ or [Database Access Controls](../../enroll-resources/database-access/rbac.mdx).
 
 ```code
-$ tsh login --proxy=<Var name="teleport.example.com:443" /> --user=myuser@example.com
+$ tsh login --proxy=<Var name="teleport.example.com" />:443 --user=myuser@example.com
````
The proxy isn't guaranteed to be on 443, so I think just drop it from here. Though I see that `teleport.example.com:443` is used throughout MCP docs as the value for the proxy.

ravicious commented on docs/pages/connect-your-client/model-context-protocol/mcp-access.mdx:
```diff
@@ -15,7 +15,9 @@ Teleport.
 
-First, sign in into your Teleport cluster using `tsh login`:
+First, sign in into your Teleport cluster using `tsh login`, assigning 
+<Var name="teleport.example.com:443" /> to the web address of the Teleport Proxy
```
```suggestion
<Var name="teleport.example.com" /> to the web address of the Teleport Proxy
```

ravicious commented on docs/pages/connect-your-client/model-context-protocol/mcp-access.mdx:
````diff
@@ -15,7 +15,9 @@ Teleport.
+Service in your cluster:
 
 ```code
 $ tsh login --proxy=<Var name="teleport.example.com:443" /> --user=myuser@example.com
````
```suggestion
$ tsh login --proxy=<Var name="teleport.example.com" /> --user=myuser@example.com
```

ravicious commented on docs/pages/enroll-resources/desktop-access/getting-started.mdx:
```diff
@@ -17,6 +17,20 @@ Community Edition will not allow connections to any of them.
+The Teleport Desktop Service proxies traffic between the Teleport Proxy Service
+and remote Windows desktops in your infrastructure. Teleport users connect to a
+Windows desktop using Teleport Connect or the Teleport Web UI. These graphical
+clients use the proprietary Teleport Desktop Protocol to send traffic to and
+receive traffic from the Teleport Desktop Service (using WebSockets), which
```
Nit: technically, Connect doesn't use WebSockets for that. I'm not sure if we need to mention the backing technology at all. ;)

Edit: Unless that's the point of this section to, uh, mention how exactly it works? Connect uses gRPC for that, but I'm just not sure if it's worth mentioning, same for WebSockets.

ptgott replied:
Thanks! I was incorrectly applying my understanding of the Web UI. I'll remove the parentheses here.

ravicious commented on docs/pages/enroll-resources/mcp-access/stdio.mdx:
```diff
@@ -112,7 +112,8 @@ EOF
 (!docs/pages/includes/create-role-using-web.mdx!)
 
-Create the Teleport user assigned the `access-mcp` role we've just created:
+Create a Teleport user assigned to the `access-mcp` role we've just created,
+assigning <Var name="my_user" /> to the name of your new Teleport user:
```
```suggestion
Create a Teleport user <Var name="my_user" /> assigned to the `access-mcp` role we've just created:
```

ravicious commented on docs/pages/enroll-resources/mcp-access/stdio.mdx:
````diff
@@ -112,7 +112,8 @@ EOF
+assigning <Var name="my_user" /> to the name of your new Teleport user:
 
 ```code
 $ tctl users add --roles=access-mcp my_user
````
```suggestion
$ tctl users add --roles=access-mcp <Var name="my_user" />
```

ptgott replied:
This line was also changed in a conflicting commit

ravicious commented on docs/pages/enroll-resources/mcp-access/troubleshooting.mdx:
````diff
@@ -25,7 +25,9 @@ spec:
 ```
 
-You can configure this user trait with `tctl`:
+You can configure this user trait with `tctl`, assigning <Var name="my_user" />
+to the name of the Teleport user you want to update:
````
```suggestion
You can configure this user trait with `tctl`, assigning it to the Teleport user <Var name="my_user" />:
```

ravicious commented on docs/pages/reference/agent-services/database-access-reference/cli.mdx:
```diff
@@ -399,10 +399,10 @@ $ tsh db exec "select 1" --db-user mysql --labels env=dev --parallel=5 --output-
 | `--db-name` | The database name to log in to. |
 | `--db-roles` | List of comma separate database roles to use for auto-provisioned user. |
-| `--dbs` |  List of comma separated target database services. Mutually exclusive with `--search` or `--labels`. |
+| `--dbs` |  List of comma separated target databases. Mutually exclusive with `--search` or `--labels`. |
```
"database services" is correct in this instance I think, otherwise the difference between `--dbs` and `-db-name` might not be clear enough.

ravicious commented on docs/pages/reference/agent-services/database-access-reference/cli.mdx:
```diff
@@ -399,10 +399,10 @@ $ tsh db exec "select 1" --db-user mysql --labels env=dev --parallel=5 --output-
 | `--search` | List of comma separated search keywords or phrases enclosed in quotations (e.g. `--search=foo,bar,"some phrase"`). |
 | `--labels` | List of comma separated labels to filter by labels (e.g. `key1=value1,key2=value2`). |
-| `--output-dir` | Directory to store command output per target database service. A summary is saved as "summary.json". |
+| `--output-dir` | Directory to store command output per target database. A summary is saved as "summary.json". |
```
Same here wrt "database service".

ptgott replied:
I've reverted changes in the table and ignored the warnings instead

ptgott committed:
Respond to ravicious feedback

- In the MCP database guide, change the Proxy Service address variable
  to include the port.
- Remove the port in the Proxy Service variable in the MCP access guide.
- Remove incorrect information in the desktop access Getting Started
  guide.
- Minor wording changes.
- Revert "database services" usage removal and ignore Vale warnings for
  the table in which we use it.
